BrowserKit/Sources/TabDataStore/TabDataStore.swift

```diff
@@ -11,11 +11,53 @@ protocol TabDataStore {
 }
 
 actor DefaultTabDataStore: TabDataStore {
+    private var lastSaveTime: Date = .distantPast
+    private let minimumSaveInterval: TimeInterval = 5
+
+    private let filePath: URL = {
+        let fileManager = FileManager.default
+        let documentsDirectory = try? fileManager.url(for: .documentDirectory,
```

Comment:
This shouldn't live in the documents directory because it shows up in the users files app.
Maybe follow the same location as the current tab store, see tabsStateArchivePath in LegacyTabManagerStore for where that lives. You can probably ignore the UITesting branch code for now, I'll create a follow up ticket to handle that use case later on.